File: Scripts/generated_codes/python/codechef_responses/180.py
'''


Solution:

'''


def subset_sum(numbers, target, partial=[]):
    s = sum(partial)

    # check if the partial sum is equals to target
    if s == target: 
        print "Yes"
    if s >= target:
        return  # if we reach the number why bother to continue

    for i in range(len(numbers)):
        n = numbers[i]
        remaining = numbers[i+1:]
        subset_sum(remaining, target, partial + [n]) 


def main():
    num_tests = int(raw_input())

    for test in range(num_tests):
        n, m = map(int, raw_input().split())
        numbers = map(int, raw_input().split())
        subset_sum(numbers, m)


if __name__ == "__main__":
    main()

File: Scripts/generated_codes/java/180.java
/*

*/
public class Main {
	public static void main(String[] args) {
		Scanner sc=new Scanner(System.in);
		int test=sc.nextInt();
		while(test-->0){
			int n=sc.nextInt();
			int m=sc.nextInt();
			int arr[]=new int[n];
			for(int i=0;i<n;i++)
				arr[i]=sc.nextInt();
			boolean flag=false;
			for(int i=0;i<n;i++){
				int sum=arr[i];
				for(int j=i+1;j<n;j++){
					sum+=arr[j];
					if(sum==m){
						flag=true;
						break;
					}
				}
				if(flag)
					break;
			}
			if(flag)
				System.out.println("Yes");
			else
				System.out.println("No");
		}
	}
}